flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -153,7 +151,7 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
     }
   }
 
-  Future<void> _startTransaction(String? name, Object? arguments) async {
+  void _startTransaction(String? name, Object? arguments) {
```

Comment:
Why did we change this? seems not related to this PR.

Reply:
One of the tests was failing, as one of the methods called here internally now returns a future. Woking on this I saw that none of the private async functions here is awaited, as it is only called internally from sync API. I'm not sure anymore why we went async here, seems redundant?

Reply:
`beginNativeFramesCollection` is async, and we want the frames from that time, if we don't `await`, this function can be executed later and the frames are not the same anymore.
So It should be `async`.
We also call `configureScope` which is now async as well, another reason to be async.